SampleApplication.py

```diff
@@ -1,40 +1,77 @@
 import AbstractApplication as Base
-from time import sleep
+from threading import Semaphore
+
 
 class SampleApplication(Base.AbstractApplication):
+    # setup our Application
+    def __init__(self):
+        super().__init__()
+
+        # Semaphores for async execution. They make sure that the action is completed.
+        self.gesture_lock = Semaphore(0)
+        self.language_lock = Semaphore(0)
+        self.text_lock = Semaphore(0)
+        self.eyecolor_lock = Semaphore(0)
+        self.name_lock = Semaphore(0)
+
+        # Attributes of our application
+        self.user_name = None
+
     def main(self):
+        # Setting language
         self.setLanguage('en-US')
+        self.language_lock.acquire()
+
+        # Change Eyecolor
+        self.setEyeColour('red')
+        self.eyecolor_lock.acquire()
+
+        # Saying something with body language
         self.sayAnimated('Hello Group 7.')
-        sleep(2)
-        # test comment
+        self.text_lock.acquire()
+
+        # Say something without moving
+        self.say('Boring hello group 7.')
+        self.text_lock.acquire()
+
+        # Do a gesture
+        # untitled-2ae403 is the ID of the Project in Choreographe.
+        # Did not know this in advance, hence the awful name.
+        self.doGesture('untitled-2ae403/behavior_1')
```

Comment:
If we put the project name, does it know what files to use automatically?

Reply:
In their sample code it seemed like it. Also, this Choreographe project is copied in the PyCharm project folder. So it should be able to find it i think.